src/main/java/com/software/modsen/rideservice/service/ServiceImpl/RideServiceImpl.java

```diff
@@ -0,0 +1,100 @@
+package com.software.modsen.rideservice.service.ServiceImpl;
+
+import com.software.modsen.rideservice.dto.request.RideRequest;
+import com.software.modsen.rideservice.dto.response.RideResponse;
+import com.software.modsen.rideservice.dto.response.RideResponseSet;
+import com.software.modsen.rideservice.exception.RideNotFoundException;
+import com.software.modsen.rideservice.mapper.RideMapper;
+import com.software.modsen.rideservice.model.Ride;
+import com.software.modsen.rideservice.model.RideStatus;
+import com.software.modsen.rideservice.repositories.RideRepository;
+import com.software.modsen.rideservice.service.RideService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.software.modsen.rideservice.util.ExceptionMessages.RIDE_NOT_FOUND;
+
+
+@Service
+@RequiredArgsConstructor
+public class RideServiceImpl implements RideService {
+
+    private final RideRepository rideRepository;
+    private final RideMapper rideMapper;
```

Comment:
Minor: usually mapping logic goes on controller layer

Reply:
Moved the mapping logic to the controller layer